core/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceParserHelper.java

```diff
@@ -99,4 +100,72 @@ public static void declareGeoFields(
         }
     }
 
+    public static <T> void declareAnyFields(
```

Comment:
I think this whole block can go if all the existing declare* methods in this class take AbstractObjectParser instead of ObjectParser as an argument. That works when getName() (see above) is mave abstract in AbstractObjectParser.